assets/src/editor/events/hocs/with-editor-event-date-entities.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External imports
+ */
+import { withSelect } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+
+const DEFAULT_OBJECT = {
+	dateEntities: [],
+};
+
+/**
+ * A HOC exposing date entities related to the provided eventEntity.
+ *
+ * @return {function} A higher order component.
+ */
+const withEditorEventDateEntities = createHigherOrderComponent(
+	withSelect(
+		( select, { eventEntity } ) => {
+			const { getRelatedEntities } = select( 'eventespresso/core' );
+			if ( isModelEntityOfModel( eventEntity, 'event' ) ) {
+				const dateEntities = getRelatedEntities(
+					eventEntity,
+					'datetime'
+				);
+				return { dateEntities };
+			}
+			return DEFAULT_OBJECT;
+		}
+	),
+	'withEditorEventDateEntities'
+);
+
+export default withEditorEventDateEntities;
```

Comment:
This component should not be needed.  `withEditorDateEntities` and `useEditorDateEntities` already exist and already return all date entities existing in the state for the editor instance (which will be for a specific event - because only the only dates in the store will be for that event).

Reply:
hmmmm... ya ok... I didn't see any logic pertaining to the current event being edited, but that would likely be the case regardless... can't think of any cases where that would be different... will change

Reply:
This is related to the work I did in creating the Editor Initialization layer (and why these hocs have the word `Editor` in them).  Essentially in the initialization layer, all the datetimes for the event are fetched and added to the state.  From that point forward any datetime in the state is related to the event being edited, so no need to get the datetimes related to the event (you can instead just get all the datetimes in the store state).